generate/unix/Makefile.config

```diff
@@ -227,22 +227,40 @@ CWARNINGFLAGS += \
     -Wold-style-definition\
     -Wno-format-nonliteral\
     -Wredundant-decls
+
+# output directory for tests below
 #
-# Per-host flags and exclusions
+# Taken from the Linux kernel
+TMPOUT = $(if $(KBUILD_EXTMOD),$(firstword $(KBUILD_EXTMOD))/).tmp_$$$$
+
+# try-run
+# Usage: option = $(call try-run, $(CC)...-o "$$TMP",option-ok,otherwise)
+# Exit code chooses option. "$$TMP" serves as a temporary file and is
+# automatically cleaned up.
+#
+# Taken from the Linux kernel
+try-run = $(shell set -e;		\
+	TMP=$(TMPOUT)/tmp;		\
+	TMPO=$(TMPOUT)/tmp.o;		\
+	mkdir -p $(TMPOUT);		\
+	trap "rm -rf $(TMPOUT)" EXIT;	\
+	if ($(1)) >/dev/null 2>&1;	\
+	then echo "$(2)";		\
+	else echo "$(3)";		\
+	fi)
+
+# Returns either the given flag if supported, or nothing
 #
-ifneq ($(ACPI_HOST), _FreeBSD)
-    CWARNINGFLAGS += \
-        -Wempty-body
+# Taken from the Linux kernel, but modified to fit our needs
+cc-option = $(call try-run, $(CC) -Werror $(1) -c -x c /dev/null -o "$$TMP",$(1),)
 
-    ifneq ($(ACPI_HOST), _APPLE)
-        CWARNINGFLAGS += \
-            -Woverride-init\
-            -Wlogical-op\
-            -Wmissing-parameter-type\
-            -Wold-style-declaration\
-            -Wtype-limits
-    endif
-endif
+CWARNINGFLAGS += $(call cc-option,-Wempty-body)
+CWARNINGFLAGS += $(call cc-option,-Wlogical-op)
+CWARNINGFLAGS += $(call cc-option,-Wmissing-parameter-type)
+CWARNINGFLAGS += $(call cc-option,-Wold-style-declaration)
+CWARNINGFLAGS += $(call cc-option,-Woverride-init)
+CWARNINGFLAGS += $(call cc-option,-Wtype-limits)
+CWARNINGFLAGS += $(call cc-option,-Wtype-clablabaldf)
```

Comment:
probably didn't mean to commit this last one, right?

Reply:
Heh, no. But shows that it works. Fixed, thanks.